### PR TITLE
fix: recent category should be shown for reactions WPB-4348

### PR DESCRIPTION
**Pull request by MarcinR**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4348" title="WPB-4348" target="_blank">StoryWPB-4348</a>  [Android] Rich text formatting should appear in the correct format inside the text field
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
This PR contains fix for a bug where after user pick emoji from reaction picker it's not added to recent category.

### CODE REVIEW

**Review by johnxnguyen — APPROVED** [review]

File: wire-ios/Wire-iOS Tests/ReferenceImages/CompleteReactionPickerViewControllerTests/testReactionPicker_withRecentReactionsSection.1.png
Comment (agisilaos) [inline wire-ios/Wire-iOS Tests/ReferenceImages/CompleteReactionPickerViewControllerTests/testReactionPicker_withRecentReactionsSection.1.png:1]: @MarcinR The category is wrong. The bar needs to be below recent category. I guess we need to set the proper index somehow like we do in `CompleteReactionViewControllerSnapshotTests`

**Review by agisilaos — CHANGES_REQUESTED** [review]

File: wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    // MARK: setUp
+
+    override func setUp() {
+        super.setUp()
+        RecentlyUsedEmojiPeristenceCoordinator.store(RecentlyUsedEmojiSection(capacity: 15))
+    }
+
+    func test_storeAndLoadEmojis() {
+        // THEN
+        XCTAssertArrayEqual(RecentlyUsedEmojiPeristenceCoordinator.loadOrCreate().emoji, [])
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift]: I think it's better to have a separate test for checking the empty storage.

File: wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+@testable import Wire
+import XCTest
+
+final class RecentlyUsedEmojiPeristenceCoordinatorTests: XCTestCase {
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift:23]: You can also add a test for adding new emoji here, no need to create a snapshot test

Comment (MarcinR) [reply wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift:23]: there is a need. Header view in `CompleteReactionViewController` looks different if there  are or not any emojis in recent category

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/CompleteReactionPicker/ReactionSectionViewController.swift [hunk 3]
```diff
@@ -55,10 +56,8 @@ final class ReactionSectionViewController: UIViewController {
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        guard let firstButton = typesByButton.first?.key else { return }
-        for button in sectionButtons {
-            button.isSelected = (button == firstButton)
-        }
+        guard let firstType = types.first else { return }
+        selectedType = firstType
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/CompleteReactionPicker/ReactionSectionViewController.swift:59]: Why don't you want to replace selectedType in the `init` method? You set a value for selectedType in init too.

Comment (MarcinR) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/CompleteReactionPicker/ReactionSectionViewController.swift:59]: setting `selectedType` updates all buttons visual properties and it's not working until view is completely loaded

Comment (KaterinaWire) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/CompleteReactionPicker/ReactionSectionViewController.swift:59]: So, can you remove it from `init` since it will be overrided every time in `viewWillAppear`?

File: wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    func test_storeAndLoadEmojis() {
+        // GIVEN
+        let emojis = [Emoji(value: "😂"), Emoji(value: "🆎"), Emoji(value: "🫥"), Emoji(value: "🐞")]
+        let emojiSection = RecentlyUsedEmojiSection(capacity: 15, elements: emojis)
+
+        // WHEN
+        RecentlyUsedEmojiPeristenceCoordinator.store(emojiSection)
+
+        // THEN
+        XCTAssertArrayEqual(RecentlyUsedEmojiPeristenceCoordinator.loadOrCreate().emoji, emojis)
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift]: Looks like `RecentlyUsedEmojiPeristenceCoordinator` should have a "emojis" property instead of "emoji" since it's an array of emojis, not just one.

File: wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+final class RecentlyUsedEmojiPeristenceCoordinatorTests: XCTestCase {
+
+    // MARK: setUp
+
+    override func tearDown() {
+        RecentlyUsedEmojiPeristenceCoordinator.store(RecentlyUsedEmojiSection(capacity: 15))
+        super.tearDown()
+    }
+
+    func test_RecentlyUsedEmoji_isEmpty() {
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift]: The name of the test doesn't follow the same style, you use uppercase and lowercase letters after the "_".
Also the name and content of the test doesn't explain why it should be empty.
I would add:

```
// WHEN
RecentlyUsedEmojiPeristenceCoordinator.store(RecentlyUsedEmojiSection(capacity: 15))
```

File: wire-ios/Wire-iOS Tests/ConversationMessageCell/CompleteReactionPickerViewControllerTests.swift [hunk 6]
```diff
@@ -36,6 +36,7 @@ final class CompleteReactionPickerViewControllerTests: BaseSnapshotTestCase {
 
     override func tearDown() {
         sut = nil
+        RecentlyUsedEmojiPeristenceCoordinator.store(RecentlyUsedEmojiSection(capacity: 15))
```
Comment (agisilaos) [inline wire-ios/Wire-iOS Tests/ConversationMessageCell/CompleteReactionPickerViewControllerTests.swift:39]: Does that need to be called in setUp?

Comment (MarcinR) [reply wire-ios/Wire-iOS Tests/ConversationMessageCell/CompleteReactionPickerViewControllerTests.swift:39]: I think it's better to use in teardown to make sure it's not influencing other tests

Comment (agisilaos) [reply wire-ios/Wire-iOS Tests/ConversationMessageCell/CompleteReactionPickerViewControllerTests.swift:39]: How about you initialize it in setUp and set it to nil in tearDown? I think that would be more structured regarding the lifecycle of the test?

File: wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift (hunk 4 above)
Comment (MarcinR) [reply wire-ios/Wire-iOS Tests/ConversationMessageCell/RecentlyUsedEmojiPeristenceCoordinatorTests.swift]: due to most dictionaries both `emojis` and `emoji` are correct form for plural. But i agree it's better for consistency in project to use one form.

File: wire-ios/Wire-iOS Tests/ConversationMessageCell/CompleteReactionPickerViewControllerTests.swift (hunk 6 above)
Comment (MarcinR) [reply wire-ios/Wire-iOS Tests/ConversationMessageCell/CompleteReactionPickerViewControllerTests.swift:39]: cant pass nil to function and it's static. I don't create instance of RecentlyUsedEmojiPeristenceCoordinator

**Review by agisilaos — APPROVED** [review]
